### PR TITLE
Added initial verification for DAB responses

### PR DESCRIPTION
**Pull request by bgy36ww**

### CODE REVIEW

File: DabTester.py [hunk 1]
```diff
@@ -19,14 +19,15 @@ def execute_cmd(self,device_id,dab_dab_request_body_topic,dab_request_body="{}")
             return 1
     
     def Execute_Test_Case(self, device_id, test_case):
-        (dab_dab_request_body_topic, dab_request_body, validate_output_function, expected_response_code)=test_case
```
Comment (akshaygarg86) [inline DabTester.py:22]: Why not expand the scope of validate_output_function and call it validate_response.

Comment (bgy36ww) [reply DabTester.py:22]: What do you mean here? You mean I call validate_output_function with validate_reaponse?

Comment (akshaygarg86) [reply DabTester.py:22]: I meant merge all validations into one function thats called validate_reaponse or any other name you like to use.

Comment (bgy36ww) [reply DabTester.py:22]: Oh, I see, that makes sense. I will try to refactor that.

Comment (bgy36ww) [reply DabTester.py:22]: I made another commit to reflect this, let me know if that's what you mean!

File: DabTester.py [hunk 2]
```diff
@@ -11,19 +11,20 @@ def __init__(self,broker):
...
-        test_result = TestResult(device_id, dab_dab_request_body_topic, dab_request_body, "UNKNOWN", "", [])
-        print("\ntesting", dab_dab_request_body_topic, " ", dab_request_body, "... ", end='', flush=True)
+        (dab_request_topic, dab_request_body, validate_output_function, expected_response_code)=test_case
+        test_result = TestResult(device_id, dab_request_topic, dab_request_body, "UNKNOWN", "", [])
+        print("\ntesting", dab_request_topic, " ", dab_request_body, "... ", end='', flush=True)
         start = datetime.datetime.now()
-        if self.execute_cmd(device_id, dab_dab_request_body_topic, dab_request_body) == 0:
+        test_result.response = self.dab_client.response()
```
Comment (akshaygarg86) [inline DabTester.py:26]: Collecting response before execution of the test?

File: dab/app_telemetry.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from schema import dab_response_validator
+from time import sleep
+from DabTester import Default_Validations
+
+
+def start(test_result, durationInMs=0,expectedLatencyMs=0):
+    dab_response_validator.validate_start_device_telemetry_response_schema(test_result.response)
+    sleep(5)
```
Comment (akshaygarg86) [inline dab/app_telemetry.py:8]: Can you recheck these sleep numbers and minimize them? Outside of app start/stop I dont think we need to wait for 5 seconds in general.

File: dab/voice.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    sleep(5)
+    if type(expectedLatencyMs) == int:
+        return YesNoQuestion(test_result, f"Can you verify the voice command ${test_result.request} has been initated?") and Default_Validations(test_result, durationInMs, expectedLatencyMs)
+    else:
+        return YesNoQuestion(test_result, expectedLatencyMs)
+
+def list(test_result, durationInMs=0,expectedLatencyMs=0):
+    dab_response_validator.validate_list_voice_response_schema(test_result.response)
+    sleep(5)
+    return Default_Validations(test_result, durationInMs, expectedLatencyMs)
```
Comment (akshaygarg86) [inline dab/voice.py:24]: can dab_response_validator be moved inside Default_Validations? Does this need to be done separately?

**Review by akshaygarg86 — COMMENTED** [review]
In general I like the idea of moving to independent validations to allow per operation customization.

File: DabTester.py (hunk 2 above)
Comment (bgy36ww) [reply DabTester.py:27]: Yea, it was error out.

Comment (akshaygarg86) [reply DabTester.py:27]: Not clear. Wondering how will the response be set before execute_cmd.

File: dab/voice.py (hunk 4 above)
Comment (bgy36ww) [reply dab/voice.py:24]: I mean, it kind of more unique to the operations themselves. Actually, I want to ask you here, when the format is not correct, should we stop the whole test or continue?

File: dab/app_telemetry.py (hunk 3 above)
Comment (bgy36ww) [reply dab/app_telemetry.py]: SG

File: dab/voice.py (hunk 4 above)
Comment (akshaygarg86) [reply dab/voice.py:24]: Just looked at schema.py. github had hidden it. The changes make sense. My opinion would be to provide a command line option for immediate fail vs continue.

**Review by akshaygarg86 — APPROVED** [review]

**Pull request merged**